src/Core/MySQL/PacketsProtocolBinary.cpp

```diff
@@ -0,0 +1,363 @@
+                        break;
+                    }
+                    case TypeIndex::UInt16: {
+                        UInt16 value = assert_cast<const ColumnVector<UInt16> &>(*col).getData()[row_num];
+                        buffer.write(reinterpret_cast<char *>(&value), 2);
+                        break;
+                    }
+                    case TypeIndex::UInt32: {
+                        UInt32 value = assert_cast<const ColumnVector<UInt32> &>(*col).getData()[row_num];
+                        buffer.write(reinterpret_cast<char *>(&value), 4);
+                        break;
+                    }
+                    case TypeIndex::UInt64: {
+                        UInt64 value = assert_cast<const ColumnVector<UInt64> &>(*col).getData()[row_num];
+                        buffer.write(reinterpret_cast<char *>(&value), 8);
+                        break;
+                    }
+                    case TypeIndex::Int8: {
+                        Int8 value = assert_cast<const ColumnVector<Int8> &>(*col).getData()[row_num];
+                        buffer.write(reinterpret_cast<char *>(&value), 1);
+                        break;
+                    }
+                    case TypeIndex::Int16: {
+                        Int16 value = assert_cast<const ColumnVector<Int16> &>(*col).getData()[row_num];
+                        buffer.write(reinterpret_cast<char *>(&value), 2);
+                        break;
+                    }
+                    case TypeIndex::Int32: {
+                        Int32 value = assert_cast<const ColumnVector<Int32> &>(*col).getData()[row_num];
+                        buffer.write(reinterpret_cast<char *>(&value), 4);
+                        break;
+                    }
+                    case TypeIndex::Int64: {
+                        Int64 value = assert_cast<const ColumnVector<Int64> &>(*col).getData()[row_num];
+                        buffer.write(reinterpret_cast<char *>(&value), 8);
+                        break;
+                    }
+                    case TypeIndex::Float32: {
+                        Float32 value = assert_cast<const ColumnVector<Float32> &>(*col).getData()[row_num];
+                        buffer.write(reinterpret_cast<char *>(&value), 4);
+                        break;
+                    }
+                    case TypeIndex::Float64: {
+                        Float64 value = assert_cast<const ColumnVector<Float64> &>(*col).getData()[row_num];
+                        buffer.write(reinterpret_cast<char *>(&value), 8);
+                        break;
+                    }
+                    case TypeIndex::Date: {
+                        UInt16 value = assert_cast<const ColumnVector<UInt16> &>(*col).getData()[row_num];
+                        LocalDate ld = LocalDate(DayNum(value));
+                        buffer.write(static_cast<char>(4)); // bytes_following
+                        auto year = ld.year();
```

Comment:
Whenever dealing with bits and bytes in C++ and not mere glue code, it is recommended to write out the actual types (e.g. `unsigned short` in l. 226) instead of `auto`.